flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/DefaultScheduler.java

```diff
@@ -225,13 +230,21 @@ protected void scheduleOrUpdateConsumersInternal(final ExecutionVertexID produce
 
 	@Override
 	public void allocateSlotsAndDeploy(final Collection<ExecutionVertexDeploymentOption> executionVertexDeploymentOptions) {
-		final Map<ExecutionVertexID, ExecutionVertexDeploymentOption> deploymentOptionsByVertex = groupDeploymentOptionsByVertexId(executionVertexDeploymentOptions);
+		final Collection<ExecutionVertexDeploymentOption> validDeploymentOptions = executionVertexDeploymentOptions.stream()
```

Comment:
Is it legal to schedule vertices in states other than `CREATED`? I wonder if this should be a `checkState`.

Nonetheless, I would extract this code into a separate method.

Reply:
Agreed. Let's throw exception for scheduling non-created vertices rather than silently deal with them.